frontend/src/app/forms/steps/education-info/education-info.component.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ *  @description
+ *   This component renders the education information step view and its actions.
+ */
+import { Component, Input } from '@angular/core';
+import { Validators } from '@angular/forms';
+import { ModalController } from '@ionic/angular';
+
+import { FormControl, FormGroup } from '../../../common/forms/forms';
+import { EducationInfoEntry, FormsData } from '../../../model/forms-data.model';
+import { EducationInfoEntryComponent } from '../education-info-entry/education-info-entry.component';
+
+@Component({
+  selector: 'form-education-info',
+  templateUrl: './education-info.component.html',
+  styleUrls: ['./education-info.component.scss'],
+})
+export class EducationInfoComponent {
+  @Input()
+  formsData: FormGroup<FormsData>;
+
+  /**
+   * Constructor
+   */
+  constructor(public modalController: ModalController) {}
+
+  /**
+   * add education info entry
+   */
+  async addEducationInfo(): Promise<void> {
+    const educationInfoAdd = new FormGroup<EducationInfoEntry>({
+      university: new FormControl('', Validators.required),
+      subject: new FormControl('', Validators.required),
+      degree: new FormControl('', Validators.required),
+      grade: new FormControl('', Validators.required),
+      graduationYear: new FormControl('', Validators.required),
+    });
+    const modal = await this.modalController.create({
+      component: EducationInfoEntryComponent,
+      componentProps: {
+        education: educationInfoAdd,
+      },
+      swipeToClose: true,
+      presentingElement: await this.modalController.getTop(),
+    });
+    /**
+     * save and cancel of education info form
+     */
+    modal.onDidDismiss().then((val) => {
+      if (val.data && val.data.canSubmitData) {
+        educationInfoAdd.markAsPristine();
```

Comment:
Why is `markAsPristine` needed here?

Reply:
hey @MBuchalik  that has to be removed. It was checked in accidentally while trying out to check whether a user modified any thing in UI. But we found another way to do this.